src/components/admin/campaign-applications/UploadedCampaignApplicationFiles.tsx

```diff
@@ -0,0 +1,43 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { AxiosError } from 'axios'
+import { useTranslation } from 'next-i18next'
+
+import { endpoints } from 'service/apiEndpoints'
+import { ApiErrors } from 'service/apiErrors'
+
+import { UploadedFile, UploadedFilesList } from 'components/common/file-upload/UploadedFilesList'
+
+import { useSession } from 'next-auth/react'
+import {
+  fetchCampaignApplicationFile,
+  useDeleteCampaignApplicationFile,
+} from 'service/campaign-application'
+import { AlertStore } from 'stores/AlertStore'
+
+type Props = {
+  campaignApplicationId: string
+  files: UploadedFile[]
+}
+
+export default function UploadedCampaignApplicationFiles({ files, campaignApplicationId }: Props) {
+  const { t } = useTranslation(['common', 'campaign-applications'])
+  const queryClient = useQueryClient()
+  const { data: session } = useSession()
+
+  const del = useMutation<unknown, AxiosError<ApiErrors>, string>({
+    mutationFn: (fileId) => useDeleteCampaignApplicationFile(session)(fileId),
+    onError: () => AlertStore.show(t('common:alerts.errorDeletingFile'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('common:files.deletedFile'), 'success')
+      queryClient.invalidateQueries([endpoints.campaignApplication.view(campaignApplicationId).url])
+    },
+  })
+
+  return (
+    <UploadedFilesList
+      files={files}
+      downloadQuery={(f) => fetchCampaignApplicationFile(f.id, session).then((r) => r.data)}
```

Comment:
maybe we should use await, insead of .then()

Reply:
Help me understand what makes the `async` way preferable here? Can we agree it does the same thing in the end or did I miss something?

Reply:
@slavcho ^^^

Reply:
Sure -  I will merge it although I don't agree with it. 

Following the project style and guidelines makes the code easier to read and understand.
CamelCase and snake_case are both good naming conventions, yet merging them is not a good idea.